external-crates/move/move-vm/profiler/src/lib.rs

```diff
@@ -0,0 +1,305 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+#[cfg(debug_assertions)]
+use move_vm_config::runtime::VMProfilerConfig;
+#[cfg(debug_assertions)]
+use once_cell::sync::Lazy;
+#[cfg(debug_assertions)]
+use serde::Serialize;
+#[cfg(debug_assertions)]
+use std::collections::BTreeMap;
+
+#[cfg(debug_assertions)]
+const MOVE_VM_PROFILER_ENV_VAR_NAME: &str = "MOVE_VM_PROFILE";
+
+#[cfg(debug_assertions)]
+static PROFILER_ENABLED: Lazy<bool> =
+    Lazy::new(|| std::env::var(MOVE_VM_PROFILER_ENV_VAR_NAME).is_ok());
```

Comment:
is this worth wrapping in a lazy? How expensive is it to read env vars?

Reply:
Honestly not entirely sure. I took this code from the VM debug tracing util.